crates/fuel-core/src/graphql_api/required_fuel_block_height_extension.rs

```diff
@@ -99,50 +130,76 @@ impl Extension for RequiredFuelBlockHeightInner {
 
         if let Some(required_block_height) = self.required_height.get() {
             if let Ok(current_block_height) = current_block_height {
-                if *required_block_height > current_block_height {
+                // Check first that the node is not out of sync
+                let minimum_block_height: BlockHeight = required_block_height
+                    .saturating_sub(self.tolerance_threshold)
+                    .into();
+                if current_block_height < minimum_block_height {
                     let (line, column) = (line!(), column!());
-                    let mut response = Response::from_errors(vec![ServerError::new(
-                        format!(
-                            "The required fuel block height is higher than the current block height. \
-                            Required: {}, Current: {}",
-                            // required_block_height: &BlockHeight, dereference twice to get the 
-                            // corresponding value as u32. This is necessary because the Display 
-                            // implementation for BlockHeight displays values in hexadecimal format.
-                            **required_block_height,
-                            // current_fuel_block_height: BlockHeight, dereference once to get the 
-                            // corresponding value as u32.
-                            *current_block_height
-                        ),
-                        Some(Pos {
-                            line: line as usize,
-                            column: column as usize,
-                        }),
-                    )]);
-
-                    response.extensions.insert(
-                        CURRENT_FUEL_BLOCK_HEIGHT.to_string(),
-                        Value::Number((*current_block_height).into()),
+                    return error_response(
+                        required_block_height,
+                        &current_block_height,
+                        line,
+                        column,
                     );
-
-                    return response
                 }
+
+                // If the node is lagging behind w.r.t. the required fuel block height,
+                // but within an acceptable interval, wait until the current block height
+                // matches the required fuel block height.
+                // TODO: Add a timeout to prevent the node from waiting indefinitely.
```

Comment:
I think you can add a timeout in 1 minute + number of blocks(in seconds) that you need to wait